pages/api/module/delete.ts

```diff
@@ -0,0 +1,18 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { Module } from "../models/module";
+
+const del = async (
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> => {
+    const { id } = req.query;
+    const module = await Module.findByIdAndDelete(id).exec();
+
+    if (!module)
```

Comment:
For returning error messages, can we follow the format of `.send({error: "Message"})`. This would make it easier for the Frontend to  parse and figure out what's wrong (should've updated you before, mb)

Similar comment for the other files as well